pkg/build/api/types.go

```diff
@@ -411,6 +411,8 @@ const (
 	// BuildConfigLabelDeprecated was used as BuildConfigLabel before adding namespaces.
 	// We keep it for backward compatibility.
 	BuildConfigLabelDeprecated = "buildconfig"
+	// BuildConfigDeletionAnnotation is an annotation that marks a BuildConfig that is being deleted
+	BuildConfigDeletionAnnotation = "openshift.io/build-config.deletion-pending"
```

Comment:
If it makes sense to have inert buildConfigs then I would vote for a field in the spec vs an annotation.